### PR TITLE
Add "Do not ask again" for empty entry confirmation

### LINKED ISSUE #8296: Add "do not ask again" for Empty Entry Confirmation in JabRef
Thank you very much everybody, but especially @ruoyu-qian for your contributions :)))
I am very happy. So far, everything works perfectly! 👌

Additional enhancements would be:

- add checkbox for `do not ask for this kind of issue again`
- add tiny `?` that explains what an 'empty entry' is. Something like: 
   - An empty entry is defined as  `@entrytype{, }`. An example would be: `@Article{,
}`. An empty entry can be found in the library-file or in biblatex source when selecting the empty entry via the entry editor.

_Originally posted by @ThiloteE in https://github.com/JabRef/jabref/issues/8096#issuecomment-968093326_

### PR DESCRIPTION
**Pull request by ruoyu-qian**
Fixes #8296, a follow-up from https://github.com/JabRef/jabref/issues/8096#issuecomment-968093326

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/preferences/GeneralPreferences.java
Comment (calixtus) [inline src/main/java/org/jabref/preferences/GeneralPreferences.java:142]: This javadoc is superfluous. Please comment on the the variable or the property, but not on every getter and setter

File: src/main/java/org/jabref/gui/JabRefFrame.java
Comment (calixtus) [inline src/main/java/org/jabref/gui/JabRefFrame.java:1194]: Why don't you use the showConfirmationDialogWithOptOutAndWait? The AlertType.CONFIRMATION is used there too. 
This looks to me like unnecessary code duplication, or didn't I see anything?

Comment (ruoyu-qian) [reply src/main/java/org/jabref/gui/JabRefFrame.java:1194]: The reason I added showCustomButtonDialogWithOptOutAndWait is because I would like to include the 'Return to JabRef' option as it has now. I believe showConfirmationDialogWithOptOutAndWait only allows for Yes/No choice, but not the third option.

File: src/main/java/org/jabref/preferences/GeneralPreferences.java
Comment (ruoyu-qian) [reply src/main/java/org/jabref/preferences/GeneralPreferences.java]: Sorry about this. I removed these comments. Let me know if you would like have any other changes.

**Comment by ThiloteE** [comment]
Would this make it into stable 5.4?

**Review by Siedlerchr — APPROVED** [review]
From my point of view it's good to go. Waiting for final approval from @calixtus

File: src/main/java/org/jabref/gui/JabRefDialogService.java
Comment (calixtus) [inline src/main/java/org/jabref/gui/JabRefDialogService.java:271]: There still seems to be some code duplication with `showConfirmationDialogWithOptOutAndWait`, which is basically now a subset of this method here. Should call this instead.
Also the method argument `type` in the method signature, but is not used in the method.

**Review by calixtus — COMMENTED** [review]
Hi, I took a look into your PR again, there is still an issue with some code duplication.
Also I tested your PR manually and it did not work as expected.

1. Create file with just ```
@Article{,
}```
2. Import as new library
3. Close again
4. Dialog shows up
5. Select return to JabRef
6. Close tab
7. Tab wont close
8. Select other tab
9. Tab cannot be selected again



Please look into these issues.

**Review by calixtus — CHANGES_REQUESTED** [review]
.

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
The tab gets blocked. Please investigate

**Comment by calixtus** [comment]
Closing this pull request due to inactivity :zzz: 
Please reopen this PR if you like to fix the bugs if you like, we would love to see this feature finally implemented.

**Comment by ThiloteE** [comment]
Another issue with the current implementation: #8645
Maybe 8645 could also be addressed, when finalizing this pull-request. Could of course also be done in a separate pull-request.

**Pull request closed without merging**
